mathics/builtin/scipy_utils/optimizers.py

```diff
@@ -1,20 +1,24 @@
 # -*- coding: utf-8 -*-
 import sys
 
+from mathics.builtin import check_requires_list
+
+from mathics.core.atoms import Number, Real
 from mathics.core.expression import Expression
 from mathics.core.evaluation import Evaluation
-from mathics.core.atoms import Number, Real
+from mathics.core.evaluators import apply_N
 from mathics.core.list import ListExpression
 from mathics.core.symbols import Symbol
 from mathics.core.systemsymbols import SymbolAutomatic, SymbolInfinity, SymbolFailed
-from mathics.core.evaluators import apply_N
+from mathics.core.utils import IS_PYPY
 
 SymbolCompile = Symbol("Compile")
 
 IS_PYPY = "__pypy__" in sys.builtin_module_names
-if IS_PYPY:
+if IS_PYPY or not check_requires_list(["scipy", "numpy"]):
```

Comment:
Maybe use utils.IS_PYPY instead defining the local version in the previos line.

Reply:
Yes, that was the intent. Forgot to remove this. 

BTW a lot of stuff in mathics.core.utils should be moved out of core since it really isn't related to core. But that's for another time.